nodejs/eks/nodegroup.ts

```diff
@@ -448,7 +450,26 @@ ${customUserData}
         userData: userdata,
     }, { parent: parent, ignoreChanges: ignoreChanges });
 
-    const workerSubnetIds = args.nodeSubnetIds ? pulumi.output(args.nodeSubnetIds) : pulumi.output(core.subnetIds).apply(ids => computeWorkerSubnets(parent, ids));
+    // Compute the worker node group subnets to use from the various approaches.
+    let workerSubnetIds: pulumi.Output<string[]> = pulumi.output([]);
+    workerSubnetIds = pulumi.all([
```

Comment:
This doesn't need to be inside of an `all/apply` if we make `core.privateSubnetIds` and `core.publicSubnetIds` optional. With that change, you can just write this:

```typescript
let workerSubnetIds: pulumi.Output<string[]>;
if (args.nodeSubnetIds !== undefined) {
    workerSubnetIds = pulumi.output(args.nodeSubnetIds);
} else if (core.privateSubnetIds !== undefined) {
    workerSubnetIds = core.privateSubnetIds;
} else if (core.publicSubnetIds !== undefined) {
    workerSubnetIds = core.publicSubnetIds;
} else {
    workerSubnetIds = pulumi.output(subnetIds).apply(ids => computeWorkerSubnets(parent, ids));
}
```

Reply:
Done